proto/server/datastore/datastore.proto

```diff
@@ -2,348 +2,329 @@ syntax = "proto3";
-message UpdateAttestedNodeEntryRequest {
-    // SPIFFE ID
-    string baseSpiffeId = 1;
-    // Serial number
-    string certSerialNumber = 2;
-    // Expiration date
-    string certExpirationDate = 3;
+message CreateNodeResolverMapEntryResponse {
+    NodeResolverMapEntry entry = 1;
 }
 
-// Represents the updated Attested node entry
-message UpdateAttestedNodeEntryResponse {
-    // Attested node entry
-    AttestedNodeEntry attestedNodeEntry = 1;
+message ListNodeResolverMapEntriesRequest {
+    string spiffe_id = 1;
 }
 
+message ListNodeResolverMapEntriesResponse {
+    repeated NodeResolverMapEntry entries = 1;
+}
 
-// Represents the Spiffe ID of the Attested node entry to delete
-message DeleteAttestedNodeEntryRequest {
-    // SPIFFE ID
-    string baseSpiffeId = 1;
+message DeleteNodeResolverMapEntryRequest {
+    NodeResolverMapEntry entry = 1;
 }
 
-// Represents the deleted Attested node entry
-message DeleteAttestedNodeEntryResponse {
-    AttestedNodeEntry attestedNodeEntry = 1;
+message DeleteNodeResolverMapEntryResponse {
+    repeated NodeResolverMapEntry entries = 1;
 }
 
-//
-//
-//
+message RectifyNodeResolverMapEntriesRequest {
+    repeated NodeResolverMapEntry entries = 1;
+}
 
-// Represents a Node resolver map entry to create
-message CreateNodeResolverMapEntryRequest {
-    // Node resolver map entry
-    NodeResolverMapEntry nodeResolverMapEntry = 1;
+message RectifyNodeResolverMapEntriesResponse {
+    repeated NodeResolverMapEntry entries = 1;
 }
 
-// Represents the created Node resolver map entry
-message CreateNodeResolverMapEntryResponse {
-    // Node resolver map entry
-    NodeResolverMapEntry nodeResolverMapEntry = 1;
+/////////////////////////////////////////////////////////////////////////////
+// AttestedNodeEntry Messages
+/////////////////////////////////////////////////////////////////////////////
+
+message AttestedNodeEntry {
+    // Node SPIFFE ID
+    string spiffe_id = 1;
+
+    // Attestation data type
+    string attestation_data_type = 2;
+
+    // Node certificate serial number
+    string cert_serial_number = 3;
+
+    // Node certificate not_after (seconds since unix epoch)
+    int64 cert_not_after = 4;
+}
+
+message CreateAttestedNodeEntryRequest {
+    AttestedNodeEntry entry = 1;
 }
 
+message CreateAttestedNodeEntryResponse {
+    AttestedNodeEntry entry = 1;
+}
 
-// Represents a Spiffe ID
-message FetchNodeResolverMapEntryRequest {
-    // SPIFFE ID
-    string baseSpiffeId = 1;
+message FetchAttestedNodeEntryRequest {
+    string spiffe_id = 1;
 }
 
-// Represents a list of Node resolver map entries for the specified Spiffe ID
-message FetchNodeResolverMapEntryResponse {
-    // List of Node resolver map entries
-    repeated NodeResolverMapEntry nodeResolverMapEntryList = 1;
+message FetchAttestedNodeEntryResponse {
+    AttestedNodeEntry entry = 1;
 }
 
+message ListAttestedNodeEntriesRequest {
+    google.protobuf.Int64Value by_expires_before = 1;
```

Comment:
Whats the advantage of using this over a regular `int64` type? I see a few other spots using similar wrapper message

Reply:
The wrappers represent optional filters. A message type is required in proto3 for conveying optionality, unless you encode the zero value as meaning "not set". I wanted to make opting into filtering explicit, thus the wrapper message.